src/OpenTelemetry.Exporter.Prometheus/PrometheusExporterOptions.cs

```diff
@@ -54,7 +54,7 @@ public IReadOnlyCollection<string> HttpListenerPrefixes
             get => this.httpListenerPrefixes;
             set
             {
-                _ = value ?? throw new ArgumentNullException(nameof(this.httpListenerPrefixes));
+                Guard.ThrowIfNull(value, nameof(this.httpListenerPrefixes));
```

Comment:
@cijothomas @alanwest Hey just FYI I think we can improve our `Guard` helpers a bit. C#10 has this thing [CallerArgumentExpression](https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-10#callerargumentexpression-attribute-diagnostics) which exists really for these types of helpers. You do something like this...

```csharp
		[DebuggerHidden]
		[MethodImpl(MethodImplOptions.AggressiveInlining)]
		public static void ThrowIfNull<T>([NotNull] T? value, [CallerArgumentExpression("value")] string? paramName = null)
			where T : class
		{
			if (value is null)
			{
				throw new ArgumentNullException(paramName, "Must not be null");
			}
		}
```

Which allows you to use it without the `nameof`:

```csharp
   Guard.ThrowIfNull(value);
```

I'm pretty sure it is just compiler magic (nothing runtime) so it _should_ work for legacy targets as well.